clang/test/SemaCXX/attr-gsl-owner-pointer.cpp

```diff
@@ -59,9 +70,21 @@ class [[gsl::Owner(int)]] AddTheSameLater{};
 class [[gsl::Owner(int)]] AddTheSameLater;
 
 class [[gsl::Owner()]] OwnerWithEmptyParameterList{};
+static_assert(__is_gsl_owner(OwnerWithEmptyParameterList), "");
 
 class [[gsl::Pointer()]] PointerWithEmptyParameterList{};
+static_assert(__is_gsl_pointer(PointerWithEmptyParameterList), "");
```

Comment:
Now it is possible to write metaprograms that make decisions based on a type being a Pointer or Owner :) Of course, people not supposed to use __ names diretly, so this is not really exposed to the user. I wonder what will the reviewers think about this.

Reply:
I copied the approach from https://reviews.llvm.org/D50119, which makes testing very clean. Seems to be not controversial there.